Add tutorial to authenticate HyP3 sdk (#598)
ASFHyP3/hyp3-docs: docs/using/authentication.md

[Earthdata Login web interface](https://urs.earthdata.nasa.gov/ "urs.earthdata.nasa.gov/" ){target=_blank} 
or by using the 
[User Tokens API](https://urs.earthdata.nasa.gov/documentation/for_users/user_token#api "urs.earthdata.nasa.gov/documentation/for_users/user_token#api" ){target=_blank}.

## Authentication in Vertex

Click the **Sign In** button in [Vertex](https://search.asf.alaska.edu/ "search.asf.alaska.edu" ){target=_blank}, 
which opens a 
[version of the Earthdata Login GUI](https://urs.earthdata.nasa.gov/oauth/authorize?response_type=code&client_id=BO_n7nTIlMljdvU6kRRB3g&redirect_uri=https://auth.asf.alaska.edu/login "ASF Authorization Login" ){target=_blank} 
customized for ASF applications. Enter your 
[Earthdata Login credentials](#earthdata-login-edl "Jump to the Earthdata Login section of this document") 
to enable access to all functionality in Vertex linked to EDL credentials, including 
[On Demand capabilities](vertex.md "Jump to the On Demand Sentinel-1 Processing in Vertex page").

![Sign In with EDL in Vertex](../images/vertex-sign-in.png "Sign In with Earthdata Login Credentials in Vertex")

> ## Authentication with HyP3 API

Author: ```suggestion
## Authentication with the HyP3 API
```

[Earthdata Login bearer token](#earthdata-login-bearer-token "Jump to the Earthdata Login Bearer Token section of this document").

### Earthdata Login Session Cookie

You can authorize the HyP3 API by having a valid Earthdata Login (asf-urs) session cookie in your web environment. 
This cookie is generated when you enter your Earthdata Login credentials in a browser, either by 
[authenticating in Vertex](#authentication-in-vertex "Jump to the Authenticating in Vertex section of this document") 
or by using the 
[Earthdata Login](#username-and-password "Jump to the EDL Username and Password section of this document") 
web interface directly.

If you do not have a valid session cookie before launching the 
[HyP3 API Swagger UI](https://hyp3-api.asf.alaska.edu/ui/ "hyp3-api.asf.alaska.edu/ui" ){target=_blank}, 
you will see an error response when submitting an API request: 

![Authentication Error](../images/authentication-error.png)

You may need to refresh the browser window with the HyP3 API Swagger UI after you authenticate using either Vertex or 
the Earthdata Login website before you will be able to submit API requests successfully.

### Earthdata Login Bearer Token

> Request an EDL token, as described in the 

Author: ```suggestion
Users can generate an EDL token, as described in the 
```

[Earthdata Login Token](#earthdata-login-token "Jump to the Earthdata Login Token section of this document") 
section of this document.

Once you have a valid EDL token: 

1. Open the [Swagger UI interface for the HyP3 API](https://hyp3-api.asf.alaska.edu/ui/ "hyp3-api.asf.alaska.edu/ui" ){target=_blank}
2. Click the **Authorize** button at the top of the Swagger UI page
3. Enter or paste your EDL token in the **Value** field of the `BearerAuth` section
4. Click the **Authorize** button in the `Available authorizations` window to apply the token

![Authorize EDL Token in API](../images/api-authorize.png "Authorize an EDL Token in the HyP3 API")

## Authentication with HyP3 Python SDK

To authenticate to the API when using the 
[HyP3 Python SDK](../using/sdk.md "hyp3-docs.asf.alaska.edu/using/sdk"), 
use the 

> [HyP3 initializer method](https://hyp3-docs.asf.alaska.edu/using/sdk_api/#hyp3_sdk.HyP3.__init__ "HyP3 SDK API Reference" ){target=_blank}.

Author: ```suggestion
[intializing the HyP3  object](https://hyp3-docs.asf.alaska.edu/using/sdk_api/#hyp3_sdk.HyP3.__init__ "HyP3 SDK API Reference" ){target=_blank}.
```

There are several options for authentication when using this method, including:

> - Adding your Earthdata Login (EDL) credentials to your local `netrc` file

Author: ```suggestion
- Adding your Earthdata Login (EDL) credentials to your local `.netrc` file
```
